fondant/component.py

```diff
@@ -142,9 +153,7 @@ def _load_or_create_manifest(self) -> Manifest:
         return Manifest.from_file(self.args.input_manifest_path)
 
     @abstractmethod
-    def transform(
-        self, args: argparse.Namespace, dataframe: dd.DataFrame
-    ) -> dd.DataFrame:
+    def transform(self, dataframe: dd.DataFrame, **kwargs) -> dd.DataFrame:
         """Abstract method for applying data transformations to the input dataframe"""
```

Comment:
Could you add an extensive docstring since this is a user facing function. You could describe the dataframe object and that the user can include additional key-word arguments